### ISSUE TITLE
[Feature Request] Show only 'ticked' search engines in popup

### ISSUE THREAD

**Issue by mherrmann3** [Robbendebiene/Gesturefy#584]
I make heavy use of your very handy gesture command "[Popop] Search Engines". Thank you a lot for this feature! (I don't need the 'Selection Search' addon anymore).

However, I have quite a few search shortcuts set up in Firefox (mostly for using them via keyword), but only a few are ticked (to not overcrowd the address bar window). So the popup is very long, as it lists all search engines irrespective of the tick.

Could you implement a switch (check box) to only list search engines in the popup that have a tick? Thanks in advance!

**Comment by Robbendebiene** [comment]
Hi, thanks for you suggestion.
Sounds reasonable to me. However extensions currently cannot access this "active" property. (see [API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/search/get)). Therefore I cannot filter them by this property.
I could recreate the user interface in Gesturefy where you can activate/deactivate the search engines one by one, but this requires a lot of work for a feature of this size. I'm not sure if I'm willing to go this way. Extending the Firefox API seems the better way to me.

**Comment by mherrmann3** [comment]
Ah, thanks for this info. I wasn't aware of the limited API. In this case you may as well close this issue.
I agree that the API should be extended, not your addon.

**Comment by Robbendebiene** [comment]
I've created a feature request (https://bugzilla.mozilla.org/show_bug.cgi?id=1706957)
Let's see how some Mozilla developers think about that :)

**Comment by mherrmann3** [comment]
Wow, thanks a lot!

**Comment by Robbendebiene** [comment]
It turns out, that this has already been requested: https://bugzilla.mozilla.org/show_bug.cgi?id=1598271
The current resolution is that Mozilla devs will probably change the search engine settings page/check mark behaviour in the near future, because of that they don't want to add any additional APIs for developers at this point.

So this issue might be solved over time by Mozilla because they consider excluding all disabled/unchecked search engines for add-ons.

**Comment by mherrmann3** [comment]
Thanks for the update, @Robbendebiene!
I interpreted it as them likely dropping the check mark in their preferences UI rework. I'm not sure if this simplification is a good thing. For me, it added a "priority" layer to the search engines. Let's see what will come - and when...

But I guess I already 'outsource' the unchecked search engines into bookmarks with their keyword in it... This will solve this issue (and is independent of their decisions).

**Comment by Robbendebiene** [comment]
> I interpreted it as them likely dropping the check mark in their preferences UI rework. I'm not sure if this simplification is a good thing. For me, it added a "priority" layer to the search engines. Let's see what will come - and when...

I agree.

> But I guess I already 'outsource' the unchecked search engines into bookmarks with their keyword in it... This will solve this issue (and is independent of their decisions).

The additional work it requires apart, this actually sounds like a good workaround without any further drawbacks!
In this case I would like to close the issue. Feel free to reopen it if Mozilla makes any relevant changes. :)

**Comment by mherrmann3** [comment]
Actually, not much work needed. I already did the 'outsourcing'; it was very fast and done in a few minutes (for ~10 search engines). I did the following procedure for each of those that I had 'unchecked':

1. Open the website for which you want to create a 'search keyword bookmark'
2. Right-click into the search field on the website --> click 'Add a keyword for this search'
3. In the popup, specify the keyword and the location for the bookmark (I created a special directory called 'Search engines')
4. Remove the search engine from the 'Search Shortcuts' list in the preferences

Easy.

I totally agree to close this issue (I was thinking to do that already with my previous post).

Thanks again for all!
